src/SQSHandler.ts

```diff
@@ -0,0 +1,322 @@
+import { Readable, Writable } from 'stream';
+import {
+  SQSClient,
+  ReceiveMessageCommand,
+  SendMessageCommand,
+  SendMessageBatchCommand,
+  DeleteMessageCommand,
+  DeleteMessageBatchCommand
+} from '@aws-sdk/client-sqs';
+
+import type {
+  Message,
+  MessageAttributeValue,
+  ReceiveMessageCommandInput,
+  SendMessageCommandInput,
+  SendMessageBatchRequestEntry,
+  DeleteMessageBatchRequestEntry
+} from '@aws-sdk/client-sqs';
+
+export type SQSOptions = { VisibilityTimeout?: number; WaitTimeSeconds?: number };
+export type ParsedMessageAttributes = Record<string, undefined | string | number | Uint8Array>;
+export type ParsedMessage = Omit<Message, 'Body' | 'MessageAttributes'> & {
+  Body: Record<string, unknown>;
+  MessageAttributes?: ParsedMessageAttributes;
+};
+export type ReceiveOptions = Omit<
+  ReceiveMessageCommandInput,
+  'QueueUrl' | 'VisibilityTimeout' | 'WaitTimeSeconds'
+>;
+export type SendOptions = Omit<
+  SendMessageCommandInput,
+  'QueueUrl' | 'MessageAttributes' | 'MessageBody'
+> & {
+  MessageAttributes?: ParsedMessageAttributes;
+};
+export type SendBatchEntry = Omit<
+  SendMessageBatchRequestEntry,
+  'MessageAttributes' | 'MessageBody'
+> & {
+  MessageBody: any;
```

Comment:
So we dont want to type it like object | string | number | ... ? instead of a any.
Or is there no sqs type for the messageBody ?

Reply:
The type is anything that you can JSON.stringify().
At that point I prefer to leave it at any rather than potentially block something that could work.

Reply:
Wouldn't it be better if we could pass a generic type to the constructor so that when we pop a message, we know it follows the typing of the generic type ?

Reply:
Btw, I made a correction to harmonize types between `ParsedMessage.Body` and `SendBatchEntry.MessageBody` which should be the same thing.
For now, they are at `any`:
https://github.com/Sagacify/sqs-handler/pull/9/files#diff-2a7469cd9fb96599d9dcba4ca696c94ad613a6d69669af7c0e1e8c955ec065cdR21

Reply:
Ok, I changed for a generic type.